pkg/virt-api/rest/interfacehotplug.go

```diff
@@ -0,0 +1,288 @@
+
+import (
+	"context"
+	"encoding/json"
+	goerrors "errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+func generateVMIInterfaceRequestPatch(vmi *v1.VirtualMachineInstance, interfaceRequest *v1.VirtualMachineInterfaceRequest) (string, error) {
+	vmiCopy := vmi.DeepCopy()
+	vmiCopy.Spec = *ApplyInterfaceRequestOnVMISpec(&vmiCopy.Spec, interfaceRequest)
+
+	oldIfacesJSON, err := json.Marshal(vmi.Spec.Domain.Devices.Interfaces)
+	if err != nil {
+		return "", err
+	}
+
+	newIfacesJSON, err := json.Marshal(vmiCopy.Spec.Domain.Devices.Interfaces)
+	if err != nil {
+		return "", err
+	}
+
+	oldNetworksJSON, err := json.Marshal(vmi.Spec.Networks)
+	if err != nil {
+		return "", err
+	}
+
+	newNetworksJSON, err := json.Marshal(vmiCopy.Spec.Networks)
+	if err != nil {
+		return "", err
+	}
+
+	testNetworks := fmt.Sprintf(`{ "op": "test", "path": "/spec/networks", "value": %s}`, string(oldNetworksJSON))
+	updateNetworks := fmt.Sprintf(`{ "op": "%s", "path": "/spec/networks", "value": %s}`, "replace", string(newNetworksJSON))
+
+	testInterfaces := fmt.Sprintf(`{ "op": "test", "path": "/spec/domain/devices/interfaces", "value": %s}`, string(oldIfacesJSON))
+	updateInterfaces := fmt.Sprintf(`{ "op": "%s", "path": "/spec/domain/devices/interfaces", "value": %s}`, "replace", string(newIfacesJSON))
+
+	patch := fmt.Sprintf("[%s, %s, %s, %s]", testNetworks, testInterfaces, updateNetworks, updateInterfaces)
+	return patch, nil
+}
+
+func generateVMInterfaceRequestPatch(vm *v1.VirtualMachine, interfaceRequest *v1.VirtualMachineInterfaceRequest) (string, error) {
+	verb := "add"
+	if len(vm.Status.InterfaceRequests) > 0 {
+		verb = "replace"
+	}
+
+	vmCopy := vm.DeepCopy()
+	if interfaceRequest.AddInterfaceOptions != nil {
+		if err := addAddInterfaceRequests(vm, interfaceRequest, vmCopy); err != nil {
+			return "", err
+		}
+	}
+
+	oldJson, err := json.Marshal(vm.Status.InterfaceRequests)
+	if err != nil {
+		return "", err
+	}
+	newJson, err := json.Marshal(vmCopy.Status.InterfaceRequests)
+	if err != nil {
+		return "", err
+	}
+
+	test := fmt.Sprintf(`{ "op": "test", "path": "/status/interfaceRequests", "value": %s}`, string(oldJson))
+	update := fmt.Sprintf(`{ "op": "%s", "path": "/status/interfaceRequests", "value": %s}`, verb, string(newJson))
+	patch := fmt.Sprintf("[%s, %s]", test, update)
+
+	return patch, nil
+}
+
+func addAddInterfaceRequests(vm *v1.VirtualMachine, ifaceRequest *v1.VirtualMachineInterfaceRequest, vmCopy *v1.VirtualMachine) error {
+	canonicalIfaceName := dynamicIfaceName(ifaceRequest)
+	for _, request := range vm.Status.InterfaceRequests {
+		if err := validateAddInterfaceRequest(request, canonicalIfaceName); err != nil {
```

Comment:
I've changed the code to ignore the request if it already in the queue - I think it is better to reject early, rather than have the controller figure it out later on.

Doing this also allowed me to remove some code, which is a good side-effect. 

Thanks for the suggestion.